python_modules/dagster-ext/dagster_ext/__init__.py

```diff
@@ -543,6 +543,80 @@ def upload_messages_chunk(self, payload: IO, index: int) -> None:
 # ########################
 
 
+class IContext(ABC):
+    """Base class for asset context implemented by AssetExecutionContext and ExtContext."""
+
+    @property
+    @abstractmethod
+    def is_asset_step(self) -> bool:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def asset_key(self) -> str:
+        """TODO."""
+
+    @property
+    @abstractmethod
+    def asset_keys(self) -> Sequence[str]:
```

Comment:
`OpExecutionContext` additionally has a `selected_asset_keys` method that may be useful to add to the interface. Interested in thoughts on that since I'm not as knowledgeable about ext use cases

Reply:
ext currently uses `OpExecutionContext.selected_asset_keys` to determine what asset keys are valid on ext-side methods-- these are passed as `ExtContextData.asset_keys`. So there is no distinction between "all" asset keys and "selected' asset keys is collapsed in ext.

But I'm not sure there even is a valid distinction in general for these contexts? I don't see an `asset_keys` property on `OpExecutionContext`. Maybe all of `OpExecutionContext`, `AssetExecutionContext`, `ExtContext` should just have an `asset_keys` method (which would imply it's on the interface too)?

Reply:
ah ok - i originally misunderstood what `asset_keys` was for the ext context. 

Based on this i think we:
* have an `asset_keys` property on `IContext` (so also on `ExtContext` and `AssetExecutionContext`). This will be the list of selected asset keys for the materialization
* deprecate `selected_asset_keys` on `AssetExecutionContext`